ui/StatusQ/src/StatusQ/Core/Utils/emojibase-parser/main.cpp

```diff
@@ -0,0 +1,198 @@
+  // hexcode (string) -> unicode
+  // order (int) -> emoji_order
+  // group (int) -> category (string)
+  // tags -> keywords (NEW)
+  // emoji -> emoji (NEW)
+  // shortname (array lookup) -> shortname + aliases (array)
+  // emoticon (array or string!) -> aliases_ascii (array)
+
+  const auto groupsMap = getGroups();
+  const auto shortNamesMap = getShortnames();
+
+  QJsonArray result;
+
+  for (int i = 0; i < size; i++) {
+    const auto value = dataArray.at(i).toObject();
+    if (value.isEmpty()) {
+      qWarning() << "Unexpected value type at index " << i << "; skipping";
+      continue;
+    }
+
+    const auto hexcode = value.value(QStringLiteral("hexcode")).toString();
+    if (!value.contains(QStringLiteral("group"))) {
+      qWarning() << "Skipping modifier emoji:" << hexcode;
+      continue;
+    }
+
+    const auto groupId = value.value(QStringLiteral("group")).toInt();
+    if (groupId == 2) {
+      qWarning() << "Skipping skin tone emoji:" << hexcode;
+      continue;
+    }
+
+    const auto emoji = value.value(QStringLiteral("emoji")).toString();
+    const auto label = value.value(QStringLiteral("label")).toString();
+    const auto order = value.value(QStringLiteral("order")).toInt();
+    const auto category = groupsMap.value(groupId);
+    const auto keywords = value.value(QStringLiteral("tags")).toArray();
+
+    auto shortnames = shortNamesMap.value(hexcode);
+    const auto shortname = !shortnames.isEmpty() ? shortnames.takeFirst() : QString();
+    const auto aliases = QJsonArray::fromStringList(shortnames);
+
+    qDebug() << "Found emoji:" << emoji << QStringLiteral("(hex: '%1' label: '%2' groupId: %3 order: %4)").arg(hexcode).arg(label).arg(groupId).arg(order);
+
+    QJsonArray aliases_ascii;
+    const auto emoticon = value.value(QStringLiteral("emoticon"));
+    if (emoticon.isString())
+      aliases_ascii.append(emoticon.toString());
+    else if (emoticon.isArray())
+      aliases_ascii = emoticon.toArray();
+
+    const auto skins = value.value(QStringLiteral("skins"));
+    result.append(transformEmoji(label, hexcode, emoji, order, category, keywords, aliases_ascii, shortname, aliases, skins.isArray() && !skins.isNull()));
+
+    if (skins.isArray()) {
+      const auto skinsArr = skins.toArray();
+      for (int j = 0; j < skinsArr.size(); j++) {
+        const auto skinnedEmoji = skinsArr.at(j).toObject();
+        const auto hexcode = skinnedEmoji.value(QStringLiteral("hexcode")).toString();
+        const auto label = skinnedEmoji.value(QStringLiteral("label")).toString();
+        const auto emoji = skinnedEmoji.value(QStringLiteral("emoji")).toString();
+        const auto order = skinnedEmoji.value(QStringLiteral("order")).toInt();
+        result.append(transformEmoji(label, hexcode, emoji, order, category, keywords, aliases_ascii, shortname, aliases, false));
+      }
+    }
+  }
+
+  QJsonDocument resultDoc(result);
+  QFile resultFile(QByteArrayLiteral("/tmp/emojiList.json"));
```

Comment:
I would write to just `emojiList.json` as /tmp is system specific.

Reply:
Fixed; also simplified the skinColor handling, removing the need for the non-search expressionfilter